src/cargo/util/context/mod.rs

```diff
@@ -634,6 +634,35 @@ impl GlobalContext {
         }
     }
 
+    /// The directory to use for intermediate build artifacts.
+    ///
+    /// Fallsback to the target directory if not specified.
+    ///
+    /// Callers should prefer [`Workspace::build_dir`] instead.
+    pub fn build_dir(&self) -> CargoResult<Option<Filesystem>> {
+        if !self.cli_unstable().build_dir {
+            return self.target_dir();
+        }
+
+        if let Some(val) = &self.build_config()?.build_dir {
+            let path = val.resolve_path(self);
```

Comment:
Compared to `pub fn target_dir()`,
- We don't have `self.build_dir` which is fine as we don't have a CLI flag
- We don't check `CARGO_BUILD_DIR` env variable as a shortcut for `CARGO_BUIILD_BUILD_DIR`.  **Should we have this shortcut?**